harvester/src/run_daily_weather.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import psycopg2.extras
+import psycopg2
+from dotenv import load_dotenv
+import logging
+import os
+import requests
+import datetime
+
+# This script fetches hourly weather data from the BrightSky API, aggregates it to daily weather data and stores it in the database
+
+# Set up logging
+logging.basicConfig()
+logging.root.setLevel(logging.INFO)
+
+# Load the environmental variables
+load_dotenv()
+
+# Check if all required environmental variables are accessible
+for env_var in [
+    "PG_SERVER",
+    "PG_PORT",
+    "PG_USER",
+    "PG_PASS",
+    "PG_DB",
+    "WEATHER_HARVEST_LAT",
+    "WEATHER_HARVEST_LNG",
+]:
+    if env_var not in os.environ:
+        logging.error("❌Environmental Variable {} does not exist".format(env_var))
+        sys.exit(1)
+
+PG_SERVER = os.getenv("PG_SERVER")
+PG_PORT = os.getenv("PG_PORT")
+PG_USER = os.getenv("PG_USER")
+PG_PASS = os.getenv("PG_PASS")
+PG_DB = os.getenv("PG_DB")
+WEATHER_HARVEST_LAT = os.getenv("WEATHER_HARVEST_LAT")
+WEATHER_HARVEST_LNG = os.getenv("WEATHER_HARVEST_LNG")
+
+# Establish database connection
+try:
+    database_connection_str = f"host='{PG_SERVER}' port={PG_PORT} user='{PG_USER}' password='{PG_PASS}' dbname='{PG_DB}'"
+    database_connection = psycopg2.connect(database_connection_str)
+    logging.info("🗄 Database connection established")
+except:
+    logging.error("❌Could not establish database connection")
+    database_connection = None
+    sys.exit(1)
+
+today = datetime.date.today()
+
+# Calculate the date one year ago
+one_year_ago = today - datetime.timedelta(days=365)
+
+# Generate a list of all dates between one year ago and today
+date_list = [
+    one_year_ago + datetime.timedelta(days=x)
+    for x in range((today - one_year_ago).days + 1)
+]
+
+
+def extract(weather_list, field):
+    return [
+        data_point[field]
+        for data_point in weather_list
+        if data_point[field] is not None
+    ]
+
+
+print(f"📅 Fetching weather data for {len(date_list)} days...")
+weather_days_in_db = []
+with database_connection.cursor() as cur:
+    cur.execute("SELECT measure_day, day_finished FROM daily_weather_data;")
+    weather_days_in_db = cur.fetchall()
+
+for date in date_list:
+    today = datetime.date.today()
+    full_day = date
+
+    existing_weather_in_db_for_this_day = [
```

Comment:
Also nitpicking. I find this list inline creation with loop and condition hard to read. 
What about:

```
# Initialize an empty list to hold the filtered data points
existing_weather_in_db_for_this_day = []

# Iterate over each data point in weather_days_in_db
for data_point_in_db in weather_days_in_db:
    # Check if the date of the first element matches full_day
    if data_point_in_db[0].date() == full_day:
        # Add this data point to the result list
        existing_weather_in_db_for_this_day.append(data_point_in_db)

```

Reply:
Also another question. Is this aggregation only to test if the data already exists in the DB? isn't there a more ergonomic way?

Reply:
```
existing_weather_in_db_for_this_day = [
        data_point_in_db
        for data_point_in_db in weather_days_in_db
        if data_point_in_db[0].date() == full_day
    ]
```

> I find this list inline creation with loop and condition hard to read.

This is a basic Python list comprehension that filters the list of weather data points to find entries that match the specific date. When familiar with Python, this is pretty straightforward to read in my opinion. I see no reason why a loop should be more precise.

> Is this aggregation only to test if the data already exists in the DB?

not sure if I understand correctly. it's not an aggregation, it's a filter on the list. I could also do a db query for each day which I want to check if it's in the database, but I decided against it because having too many db calls is also not efficient.